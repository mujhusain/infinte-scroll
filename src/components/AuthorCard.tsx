import * as React from 'react';
import {
  Avatar,
  Divider,
  Card,
  Title,
  Paragraph,
  Chip,
} from 'react-native-paper';
import {StyleSheet, View} from 'react-native';
const LeftContent = (props:any) => <Avatar.Icon {...props} icon="folder" />;

type propTypes ={
  title:string, url:string, created_at:string, tags:string[], author:string
}

const AuthorCard = ({title, url, created_at, tags, author}:propTypes) => {
  return (
    <View style={styles.card}>
      <Card elevation={2}>
        <Card.Title title={author} subtitle={url} left={LeftContent} />
        <Card.Content>
          <Title>Title: {title}</Title>
          <Paragraph>{created_at}</Paragraph>
        </Card.Content>
        <View style={styles.tags}>
          {tags?.map((item,index) => (
            <Chip key={index} icon="tag-multiple" onPress={() => console.log('Pressed')}>
              {item}
            </Chip>
          ))}
        </View>
      </Card>
      <Divider />
    </View>
  );
};

export default AuthorCard;

const styles = StyleSheet.create({
  card: {
    marginBottom: 10,
    marginHorizontal: 5,
  },
  tags: {
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    marginBottom: 10,
    marginTop: 10,
  },
});
