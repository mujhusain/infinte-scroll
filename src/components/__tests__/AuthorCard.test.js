
import 'react-native';
import React from 'react';
import {render, screen, fireEvent} from '@testing-library/react-native';

// Note: test renderer must be required after react-native.
import {create, act, renderer}  from 'react-test-renderer';
import AuthorCard from '../AuthorCard';

// SnapShots
test('renders correctly', () => {
  const tree = create(<AuthorCard />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('<AuthorCard /> Basics', () => {
    it('has 2 child', async () => {
        let tree
    act(()=>{
       tree = create(<AuthorCard />)
    })
    expect(tree.toJSON().children.length).toBe(2);
  });

  it("Component should have one test text",()=>{
    render(<AuthorCard title='test' tags={['tag1',"tag2"]} />);
    const title=screen.getByText(/Title: test/i)
    expect(title).toBeTruthy();
  })

})

