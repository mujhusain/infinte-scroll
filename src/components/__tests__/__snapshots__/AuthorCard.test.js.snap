// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 10,
      "marginHorizontal": 5,
    }
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "rgba(0, 0, 0, 0.12)",
        "borderRadius": 4,
        "elevation": 2,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.75,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexGrow": 1,
          "flexShrink": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingLeft": 16,
            },
            Object {
              "minHeight": 72,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "height": 40,
                "justifyContent": "center",
                "marginRight": 16,
                "width": 40,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#6200ee",
                  "borderRadius": 20,
                  "height": 40,
                  "width": 40,
                },
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {},
              ]
            }
          >
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontSize": 24,
                  },
                  Array [
                    Object {
                      "lineHeight": 24,
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              ó°‰‹
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              },
            ]
          }
        />
        <View />
      </View>
      <View
        style={
          Array [
            Object {
              "paddingHorizontal": 16,
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "textAlign": "left",
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "fontFamily": "System",
                  "fontWeight": "500",
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "fontSize": 20,
                    "letterSpacing": 0.15,
                    "lineHeight": 30,
                    "marginVertical": 2,
                  },
                  undefined,
                ],
              ],
            ]
          }
        >
          Title: 
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "textAlign": "left",
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "fontSize": 14,
                    "letterSpacing": 0.25,
                    "lineHeight": 20,
                    "marginVertical": 2,
                  },
                  undefined,
                ],
              ],
            ]
          }
        />
      </View>
      <View
        index={2}
        siblings={
          Array [
            "withTheme(Card.Title)",
            "Card.Content",
            "View",
          ]
        }
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-evenly",
            "marginBottom": 10,
            "marginTop": 10,
          }
        }
        total={3}
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.12)",
          "height": 0.5,
        },
        undefined,
        undefined,
      ]
    }
  />
</View>
`;
